crates/sparrow-instructions/src/evaluators/list/index.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::Context;
+use arrow::array::{Array, ArrayRef, AsArray, Int32Array, Int64Array, ListArray};
+use arrow::buffer::OffsetBuffer;
+
+use arrow_schema::DataType;
+use itertools::Itertools;
+use sparrow_plan::ValueRef;
+use std::sync::Arc;
+
+use crate::{Evaluator, EvaluatorFactory, StaticInfo};
+
+/// Evaluator for `index` on lists.
+///
+/// Retrieves the value at the given index.
+#[derive(Debug)]
+pub(in crate::evaluators) struct IndexEvaluator {
+    index: ValueRef,
+    list: ValueRef,
+}
+
+impl EvaluatorFactory for IndexEvaluator {
+    fn try_new(info: StaticInfo<'_>) -> anyhow::Result<Box<dyn Evaluator>> {
+        let input_type = info.args[1].data_type.clone();
+        match input_type {
+            DataType::List(t) => anyhow::ensure!(t.data_type() == info.result_type),
+            other => anyhow::bail!("expected list type, saw {:?}", other),
+        };
+
+        let (index, list) = info.unpack_arguments()?;
+        Ok(Box::new(Self { index, list }))
+    }
+}
+
+impl Evaluator for IndexEvaluator {
+    fn evaluate(&mut self, info: &dyn crate::RuntimeInfo) -> anyhow::Result<ArrayRef> {
+        let list_input = info.value(&self.list)?.array_ref()?;
+        let index_input = info.value(&self.index)?.primitive_array()?;
+
+        let result = list_get(&list_input, &index_input)?;
+        Ok(Arc::new(result))
+    }
+}
+
+/// Given a `ListArray` and `index` array of the same length return an array of the values.
+fn list_get(list: &ArrayRef, indices: &Int64Array) -> anyhow::Result<ArrayRef> {
+    anyhow::ensure!(list.len() == indices.len());
+
+    let list = list.as_list();
+    let take_indices = list_indices(list, indices)?;
+    arrow::compute::take(list.values(), &take_indices, None).context("take in get_map")
+}
+
+/// Gets the indices in the list where the values are at the index within each list.
+fn list_indices(list: &ListArray, indices: &Int64Array) -> anyhow::Result<Int32Array> {
+    let offsets = list.offsets();
+    Ok(accessible_array_list_indices(offsets, indices))
+}
+
+/// Generic implementation of `list_indices` for arrays implementing `ArrayAccessor`.
+fn accessible_array_list_indices(offsets: &OffsetBuffer<i32>, indices: &Int64Array) -> Int32Array {
```

Comment:
Change the name and comment -- it no longer uses `ArrayAccesor`. In fact, this can probably just be merged with `list_indices` since all that does is call this method.